tasks/compile-stylesheets.mjs

```diff
@@ -1,130 +1,103 @@
 
-import { destination, isDev, isDist, isPackage } from './task-arguments.mjs'
+import { writeAsset } from './compile-assets.mjs'
+import { destination, isDist, isPackage, isPublic } from './task-arguments.mjs'
 
-const sass = gulpSass(nodeSass)
+// Sass renderer
+const sassRender = promisify(render)
 
 /**
  * Compile Sass to CSS task
  *
- * @returns {import('stream').Stream} Output file stream
+ * @returns {Promise<void>}
  */
-export function compileStylesheets () {
-  const destPath = isPackage
-    ? join(destination, 'govuk')
-    : destination
-
-  // Release distribution
-  if (isDist) {
-    return merge(
-      compileStylesheet(
-        gulp.src(`${slash(paths.src)}/govuk/all.scss`, {
-          sourcemaps: true
-        })
-      )
-        .pipe(rename({
-          basename: 'govuk-frontend'
-        })),
-
-      compileStylesheet(
-        gulp.src(`${slash(paths.src)}/govuk/all-ie8.scss`, {
-          sourcemaps: true
-        })
-      )
-        .pipe(rename({
-          basename: 'govuk-frontend-ie8'
-        }))
-    )
-      .pipe(rename({
-        suffix: `-${pkg.version}`,
-        extname: '.min.css'
-      }))
-
-      .pipe(gulp.dest(slash(destPath), {
-        sourcemaps: '.'
-      }))
-  }
+export async function compileStylesheets () {
+  const importEntries = await getImportEntries()
 
-  // Review application
-  return merge(
-    compileStylesheet(
-      gulp.src(`${slash(paths.app)}/**/[!_]*.scss`, {
-        sourcemaps: true
-      }))
-  )
-    .pipe(rename({
-      extname: '.min.css'
-    }))
-
-    .pipe(gulp.dest(slash(destPath), {
-      sourcemaps: '.'
-    }))
+  try {
+    await Promise.all(importEntries.map(compileStylesheet))
+  } catch (cause) {
+    throw new PluginError('compile:scss', cause)
+  }
 }
 
 compileStylesheets.displayName = 'compile:scss'
 
 /**
  * Compile Sass to CSS helper
  *
- * @param {import('stream').Stream} stream - Input file stream
- * @param {import('node-sass').Options} [options] - Sass options
- * @returns {import('stream').Stream} Output file stream
+ * @param {AssetEntry} assetEntry - Asset entry
  */
-function compileStylesheet (stream, options = {}) {
-  return stream
-    .pipe(plumber(errorHandler(stream, 'compile:scss')))
-    .pipe(sass({
-      ...options,
-
-      // Resolve @imports via
-      includePaths: [
-        join(paths.node_modules, 'govuk_frontend_toolkit/stylesheets'),
-        paths.node_modules
-      ]
-    }))
-    .pipe(postcss())
-    .pipe(plumber.stop())
+export async function compileStylesheet ([modulePath, { srcPath, destPath }]) {
+  const moduleSrcPath = join(srcPath, modulePath)
+  const moduleDestPath = join(destPath, getPathByDestination(modulePath))
+
+  // Render Sass
+  const result = await sassRender({
+    file: moduleSrcPath,
+    outFile: moduleDestPath,
+
+    // Enable source maps
+    sourceMap: true,
+    sourceMapContents: true,
+
+    // Resolve @imports via
+    includePaths: [
+      join(paths.src, 'govuk'),
```

Comment:
Let's take this line out, just in case it causes something unexpected to happen - can always review later.

Reply:
All sorted, yep happy to wait. Thank you 👍